crates/environ/src/builtin.rs

```diff
@@ -1,7 +1,8 @@
 /// Helper macro to iterate over all builtin functions and their signatures.
 #[macro_export]
 macro_rules! foreach_builtin_function {
-    ($mac:ident) => {
+    ($mac:ident, exported = $exported:literal) => {
+        #[wasmtime_versioned_export_macros::versioned_foreach_builtin_function($exported)]
```

Comment:
If possible I think it'd be best to avoid the macro here and instead apply it on usages. For example when the `impl_*` symbols are defined they'd have `#[versioned_fn]`, but otherwise this macro wouldn't be tampered with and the normal identifiers would be used everywhere.